library/include/borealis/core/touch/gesture_recognizer.hpp

```diff
@@ -0,0 +1,50 @@
+/*
+    Copyright 2021 XITRIX
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include <functional>
+
+#include <borealis/core/input.hpp>
+
+namespace brls
+{
+
+class View;
+
+enum class GestureState
+{
+    INTERRUPTED,
+    UNSURE,
+    START,
+    STAY,
+    END,
+    FAILED,
+};
+
+class GestureRecognizer
```

Comment:
Can you add documentation for the class as well as for every public method please?

Reply:
I'm not good at documentation, but I've done my best. Hope it's ok.